parsers/verilog.c

```diff
@@ -420,25 +420,49 @@ static short hasSimplePortList (verilogKind kind)
 	}
 }
 
-static tokenInfo *newToken (void)
+static void clearToken (tokenInfo *token)
 {
-	tokenInfo *const token = xMalloc (1, tokenInfo);
-	token->kind = K_UNDEFINED;
-	token->name = vStringNew ();
+	token->kind = K_UNDEFINED;	// to be set by updateKind()
+	vStringClear (token->name);
 	token->lineNumber = getInputLineNumber ();
 	token->filePosition = getInputFilePosition ();
 	token->scope = NULL;
 	token->nestLevel = 0;
 	token->lastKind = K_UNDEFINED;
-	token->blockName = vStringNew ();
-	token->inheritance = vStringNew ();
+	vStringClear (token->blockName);
+	vStringClear (token->inheritance);
 	token->prototype = false;
 	token->classScope = false;
 	token->parameter = false;
 	token->hasParamList = false;
+}
+
+static tokenInfo *newToken (void)
+{
+	tokenInfo *const token = xMalloc (1, tokenInfo);
+	token->name = vStringNew ();
+	token->blockName = vStringNew ();
+	token->inheritance = vStringNew ();
+	clearToken(token);
 	return token;
 }
 
+static tokenInfo *dupToken (tokenInfo *token)
+{
+	tokenInfo *dup = newToken ();
+	vStringDelete (dup->name);
+	vStringDelete (dup->blockName);
+	vStringDelete (dup->inheritance);
+	*dup = *token;
+	dup->name = vStringNew ();
```

Comment:
The three fields are deleted, then allocated. Smaller overhead but how about using the following code instead?
```
tokenInfo backup = *dup
*dup = *token;
dup->name = backup.name;
dup->blockName = backup.blockName;
dup->inheritance = backup.inheritance;
// ...copy the contents.
```

Reply:
??? I'm confused. Is this change is needed for generating a vcxproj file?

Reply:
Sorry for your confusion.

Last week I setup a native Linux box instead of Windows 10 WSL Ubuntu I have been using. I pushed the PR from it.
It seems that I made a mistake.